### PR TITLE
WIP/On-Hold: Set etc/hosts file to resolve dnsname

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
Assisted service sets a custom dns name for service base url for onprem use by adding all the ips of the assisted service node's into /etc/hosts file of the agent machines. Once the agent machine reboots during the cluster installation, the entries in etc/hosts file on agent nodes are lost. This fix adds the required entries back into the agent's etc/host file via ignition config.

### CODE REVIEW

File: src/installer/installer.go [hunk 1]
```diff
@@ -412,3 +419,12 @@ func (i *installer) setHostnameInIgnition(ignitionPath string) error {
 	}
 	return i.ops.SetFileInIgnition(ignitionPath, "/etc/hostname", fmt.Sprintf("data:,%s", i.Hostname), 420)
 }
+
+func (i *installer) setEtcHostInIgnition(ignitionPath string) error {
```
Comment (romfreiman) [inline src/installer/installer.go:423]: @carbonin if I would you I would not allow this to be merged :)

**Comment by carbonin** [comment]
re: https://github.com/openshift/assisted-installer/pull/40/files#r490447411

There is a plan in https://issues.redhat.com/browse/MGMT-2221 to move the host-side ignition customization to the assisted-service rather than making these changes on the node.

Where are these ips coming from? If you link to the source I'll make sure they're included in the ignition that will get downloaded.
Alternatively, if this is an urgent fix, we can make the change here first and I'll move this code over to the service in my refactoring.

**Comment by carbonin** [comment]
Spoke with @pawanpinjarkar and we decided to move this to assisted-service.
1. This is not host-specific so this addition can, and should, be applied to both the master and worker ignition files in their current state
2. As @romfreiman mentioned, I'm going to be moving all of this ignition-munging logic into assisted-service soon so it would be wasted effort to make this change here then have to move it to assisted-service anyway.

The proposed change would be to build upon https://github.com/openshift/assisted-service/pull/185 by adding the `/etc/hosts` file to both `master.ign` and `worker.ign` after they are created, and before they are uploaded to S3. This can be done in parallel with the work I'm doing to move to host-based ignition files as the change would be applied identically, no matter which is merged first, the conflicts should be simple to resolve.

**Comment by eranco74** [comment]
@pawanpinjarkar I'm closing this PR

**Comment by pawanpinjarkar** [comment]
Created a new PR https://github.com/openshift/assisted-installer/pull/45 which is still needed after @carbonin completes his ignition changes.

**Pull request closed without merging**